x/leveragelp/keeper/begin_blocker.go

```diff
@@ -15,47 +16,41 @@ func (k Keeper) BeginBlocker(ctx sdk.Context) {
 	// check if epoch has passed then execute
 	epochLength := k.GetEpochLength(ctx)
 	epochPosition := k.GetEpochPosition(ctx, epochLength)
+	params := k.GetParams(ctx)
 
-	if epochPosition == 0 { // if epoch has passed
-		pools := k.GetAllPools(ctx)
+	if epochPosition == 0 && params.FallbackEnabled { // if epoch has passed
+		pageReq := &query.PageRequest{
+			Limit:      uint64(params.NumberPerBlock),
+			CountTotal: true,
+		}
+		offset, _ := k.GetOffset(ctx)
+		pageReq.Offset = offset
+		positions, _, err := k.GetPositions(ctx, pageReq)
+		if err != nil {
+			ctx.Logger().Error(errors.Wrap(err, fmt.Sprintf("error fetching paginated positions")).Error())
+			return
```

Comment:
We should have panic here. Any error that shouldn't happen should directly stop the chain (though it might not happen even with panic)